lib/datasource/crate/index.ts

```diff
@@ -170,18 +170,53 @@ async function fetchRegistryInfo(
     }
 
     const cacheKey = `crate-datasource/registry-clone-path/${registryUrl}`;
+    const cacheKeyForError = `crate-datasource/registry-clone-path/${registryUrl}/error`;
 
-    let clonePath: string = memCache.get(cacheKey);
-    if (!clonePath) {
+    // We need to ensure we don't run `git clone` in parallel. Therefore we store
+    // a promise of the running operation in the mem cache, which in the end resolves
+    // to the file path of the cloned repository.
+
+    const clonePathPromise: Promise<string> | null = memCache.get(cacheKey);
+    let clonePath: string;
+
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    if (clonePathPromise) {
```

Comment:
```suggestion
    if (!!clonePathPromise) {
```
What about this?

Reply:
Wouldn't pass lints.

![image](https://user-images.githubusercontent.com/1683034/110332351-84305180-8020-11eb-88b9-582048e1f010.png)